### PR TITLE
Add Brave Player button to navigation bar

### LINKED ISSUE #8326: Brave Player button on navigation bar
### Description: 
Show a Brave Player icon when visiting YouTube

https://www.figma.com/file/4tLGBHZNP8dL0vvphXpsd3/Brave-Player-for-YouTube?type=design&node-id=1-25&mode=design&t=658K8xsuuPbmPOG6-0

### Steps to Reproduce 
  1. Go to YouTube
  2. Click on a video
  3. Should see brave-player icon (only shows up after dismissing the YouTube warning)

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes

This pull request fixes #8326

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

See issue for STR

## Screenshots:

Screenshot 2023-10-26 at 9 09 07 PM
Screenshot 2023-10-26 at 9 09 33 PM

https://github.com/brave/brave-ios/assets/909331/d3de567f-215c-4002-8c05-263d50dac452

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift [hunk 1]
```diff
@@ -81,6 +84,21 @@ class TabLocationView: UIView {
...
     }
   }
+  
+  private func updateBravePlayerIcon() {
+    guard let url = url else {
+      bravePlayerButton.isHidden = true
+      return
+    }
+    
+    if ShieldPreferences.hasSeenAntiAdBlockWarning.value, let etldP1 = url.baseDomain, let components = URLComponents(url: url, resolvingAgainstBaseURL: false) {
```
Comment (soner-yuksel) [inline Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift:94]: So is it correct user can only see Brave Player if user pressed "Adjust Shields For Me" button on youtube pop-over ?

**Comment by diracdeltas** [comment]
doesn't this require a security review? @stoletheminerals

**Comment by diracdeltas** [comment]
reminder again that the youtube player work on iOS still needs a security review.

**Comment by cuba** [comment]
All combined here: https://github.com/brave/brave-ios/pull/8416

File: Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift (hunk 1 above)
Comment (cuba) [reply Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift:94]: no, the user always sees brave player icon. The adjust shields is just informing the user they should disable shields for YouTube

**Pull request closed without merging**